### PR TITLE
ceph-volume: dmcrypt support for lvm

### PR DESCRIPTION
**Pull request by alfredodeza**
Adds the `--dmcrypt` flag for prepare and create, enabling encryption on OSD devices.

Updates the docs, and adds the corresponding dmcrypt testing scenarios (all passing against this pr https://github.com/ceph/ceph-ansible/pull/2340 )

Fixes: http://tracker.ceph.com/issues/22619

### CODE REVIEW

File: src/ceph-volume/ceph_volume/tests/functional/lvm/xenial/filestore/dmcrypt/group_vars/all [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+
+ceph_dev: True
+cluster: ceph
+public_network: "192.168.3.0/24"
+cluster_network: "192.168.4.0/24"
```
Comment (andrewschoen) [inline src/ceph-volume/ceph_volume/tests/functional/lvm/xenial/filestore/dmcrypt/group_vars/all]: I don't see ``dmcrypt: True`` set anywhere here.

File: src/ceph-volume/ceph_volume/tests/functional/lvm/centos7/bluestore/dmcrypt/group_vars/all [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+
+dmcrypt: True
+ceph_dev: True
+cluster: ceph
```
Comment (andrewschoen) [inline src/ceph-volume/ceph_volume/tests/functional/lvm/centos7/bluestore/dmcrypt/group_vars/all]: Should we have at least one of these tests use a different cluster name?

File: src/ceph-volume/ceph_volume/tests/functional/lvm/xenial/filestore/dmcrypt/group_vars/all (hunk 1 above)
Comment (alfredodeza) [reply src/ceph-volume/ceph_volume/tests/functional/lvm/xenial/filestore/dmcrypt/group_vars/all]: I was testing you

File: src/ceph-volume/ceph_volume/tests/functional/lvm/centos7/bluestore/dmcrypt/group_vars/all (hunk 2 above)
Comment (alfredodeza) [reply src/ceph-volume/ceph_volume/tests/functional/lvm/centos7/bluestore/dmcrypt/group_vars/all]: I don't know that would trigger anything different though. Or you mean for debugging?

Comment (andrewschoen) [reply src/ceph-volume/ceph_volume/tests/functional/lvm/centos7/bluestore/dmcrypt/group_vars/all]: I was just wondering if it'd trigger some command that might have needed ``--cluster`` for any reason. If you don't think it's needed then no worries.

Comment (alfredodeza) [reply src/ceph-volume/ceph_volume/tests/functional/lvm/centos7/bluestore/dmcrypt/group_vars/all]: dios mio man

I thought this was `cluster: test` in here, not `cluster: ceph`
dios_mio_man

Comment (alfredodeza) [reply src/ceph-volume/ceph_volume/tests/functional/lvm/centos7/bluestore/dmcrypt/group_vars/all]: Custom cluster are no longer supported. 

http://lists.ceph.com/pipermail/ceph-users-ceph.com/2017-November/022244.html

Lets party

**Comment by leseb** [comment]
@alfredodeza is zap working with dmcrypt? Thanks.

**Pull request merged**